integrations/gmail/hub.md

```diff
@@ -10,14 +10,172 @@ To set up the Gmail integration using OAuth, click the authorization button and
+
+   - For example, you can generate a secure string using one the following commands:
+
+     ```bash
+     # Using openssl:
+     openssl rand -hex 32
+
+     # Using nodejs:
+     node -e "console.log(require('crypto').randomBytes(32).toString('hex'))"
+
+     # Using Python 3:
+     python3 -c "import secrets; print(secrets.token_hex(32))"
+     ```
+
+   - You may also use an online password generator to create a secure string:
+     - https://bitwarden.com/password-generator/
+     - https://1password.com/password-generator
+
+2. Copy the **shared secret** for use in the next steps.
+
+#### Creating a Pub/Sub subscription
+
+1. In the Google Cloud Console, navigate to the `Pub/Sub` section.
+2. Click on `Subscriptions` in the left sidebar.
+3. Click the `Create subscription` button.
+4. Enter a name for the subscription in the `Subscription ID` field.
+5. Select the topic you created earlier from the `Topic` dropdown.
+6. Under `Delivery type`, select `Push`.
+7. Enter your integration's Botpress-provided webhook URL in the `Endpoint URL` field. To this URL, append `?shared_secret=`, followed by the shared secret you generated earlier.
+   > For example, if your integration's webhook URL is `https://webhook.botpress.cloud/57fcfb04-51fd-4381-909a-10e6ae53d310`, the endpoint URL would be `https://webhook.botpress.cloud/57fcfb04-51fd-4381-909a-10e6ae53d310?shared_secret=your_shared_secret`.
+8. Check the `Enable authentication` checkbox.
+9. Enter the service account email address you created earlier in the `Service account` field.
+10. Enter the shared secret you generated earlier in the `Audience` field.
+11. Under `Expiration period`, select `Never expire`.
+12. Under `Acknowledgement deadline`, enter `60` seconds.
+13. Under `Retry policy`, select `Retry after exponential backoff delay`. Set the minimum backoff to `60` second and the maximum backoff to `600` seconds.
+14. Click the `Create` button to create the subscription.
+
+#### Authorizing the OAuth application
+
+1. On Gmail, log in to the Google account you want to use with the Gmail integration.
+2. Once logged in, go to the following URL in your browser:
+
+   ```
+   https://accounts.google.com/o/oauth2/auth?response_type=code&scope=https://www.googleapis.com/auth/gmail.readonly%20https://www.googleapis.com/auth/gmail.send&access_type=offline&prompt=consent&redirect_uri=https://botpress.com&client_id=
+   ```
+
+   > Make sure to add your OAuth **client ID** to the end of the URL. For example, if your OAuth client ID is `abcd`, the URL should end with `&client_id=abcd`.
+
+3. Follow the on-screen instructions to authorize the OAuth application with your personal Gmail account.
+4. You will be redirected to botpress.com. **Do not close this page**.
```

Comment:
Note to reviewers: I originally used `https://[::1]` (localhost) as redirect uri, so that the authorization code couldn't get leaked by mistake. This appeared to work fine: GCP let me use localhost as redirect uri, and it did indeed redirect me to localhost after going through the OAuth process. However, I could not get it to exchange the authorization token for a refresh token: this endpoint expects an actual domain name as the redirect URI. The next best thing, IMO is to redirect to botpress.com or webhook.botpress.cloud. This does nothing, but the user is still able to get the authorization code from the `code=` query parameter.
Another option could be to redirect to a dummy domain on the [.internal](https://www.icann.org/en/board-activities-and-meetings/materials/approved-resolutions-special-meeting-of-the-icann-board-29-07-2024-en#section2.a) TLD, which ICANN reserves for private network use. However, if we were to redirect to something like `localhost.internal`, this could leak the authorization code on the local network. For instance, `localhost.internal` could be defined in the DNS resolution table of the bot owner's local DNS resolver, although this authorization code couldn't be used without also knowing the OAuth client secret.

Reply:
side-note: this issue could be alleviated if we could display the oauth authorization button, but CLS-2449 makes it impossible currently

Reply:
@pascal-botpress added a comment to the same CLS that may help you temporarily